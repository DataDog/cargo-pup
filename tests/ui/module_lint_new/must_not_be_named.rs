//@compile-flags: --crate-name test_must_not_be_named
//@compile-flags: --crate-type lib

// These modules are allowed (don't match the bad_* pattern)
pub mod good_module {
    pub fn valid_function() {
        println!("This is an acceptable module name");
    }
}

pub mod another_good_module {
    pub fn another_function() {
        println!("This is also an acceptable module name");
    }
}

// These modules match the forbidden pattern and should trigger errors
pub mod bad_module { //~ ERROR: Module must not match pattern 'bad_*'
    pub fn invalid_function() {
        println!("This module has a forbidden name pattern");
    }
}

pub mod bad_name_module { //~ ERROR: Module must not match pattern 'bad_*'
    pub fn another_invalid_function() {
        println!("This module also has a forbidden name pattern");
    }
} 