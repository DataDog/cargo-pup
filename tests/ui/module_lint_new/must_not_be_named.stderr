error: Module must not match pattern 'bad_*'
  --> tests/ui/module_lint_new/must_not_be_named.rs:20:1
   |
LL | / pub mod bad_module {
LL | |     pub fn invalid_function() {
LL | |         println!("This module has a forbidden name pattern");
LL | |     }
LL | | }
   | |_^
   |
   = help: Choose a name that doesn't match this pattern
   = note: Applied by cargo-pup rule 'must_not_be_named_test'.
   = note: `#[deny(module_must_not_be_named)]` on by default

error: Module must not match pattern 'bad_*'
  --> tests/ui/module_lint_new/must_not_be_named.rs:26:1
   |
LL | / pub mod bad_name_module {
LL | |     pub fn another_invalid_function() {
LL | |         println!("This module also has a forbidden name pattern");
LL | |     }
LL | | } 
   | |_^
   |
   = help: Choose a name that doesn't match this pattern
   = note: Applied by cargo-pup rule 'must_not_be_named_test'.

error: aborting due to 2 previous errors

